### PR TITLE
[DUOS-2171][risk=no] Replace set-output in action

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses

https://broadworkbench.atlassian.net/browse/DUOS-2171

See also: 
* https://github.com/DataBiosphere/consent-ontology/pull/761
* https://github.com/DataBiosphere/duos-ui/pull/1996

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -21,26 +21,26 @@ jobs:
...
         ENVIRONMENT_TAG=""
         if ${{ github.event_name == 'pull_request'}}; then
-          ENVIRONMENT_TAG="${REGISTRY_HOST}/${GOOGLE_PROJECT}/${SERVICE_NAME}:pr-${{ steps.short-sha.outputs.sha }}"
+          ENVIRONMENT_TAG="${REGISTRY_HOST}/${GOOGLE_PROJECT}/${SERVICE_NAME}:pr-${{ env.sha }}"
         elif ${{github.event_name == 'push' }}; then
           ENVIRONMENT_TAG="${REGISTRY_HOST}/${GOOGLE_PROJECT}/${SERVICE_NAME}:dev"
         fi
-        echo ::set-output name=sha-tag::$SHA_TAG
-        echo ::set-output name=environment-tag::$ENVIRONMENT_TAG
+        echo "sha-tag=$SHA_TAG" >> $GITHUB_ENV
```
Comment (nmalfroy) [inline .github/workflows/build.yaml]: Not asking as a change request but just curiosity: any reason you changed to `GITHUB_ENV` as opposed to `GITHUB_OUTPUT` (I think the later would avoid the changes like on line 33)?  I actually think it makes more sense to do what you did and change to GITHUB_ENV (seems a lot easier to deal with than worry about which step an output belongs to) but more for my own education

**Review by nmalfroy — APPROVED** [review]
lgtm!

File: .github/workflows/build.yaml (hunk 1 above)
Comment (rushtong) [reply .github/workflows/build.yaml]: I think I was using [this](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions) as a guide but yeah, I could go either way here.

**Review by connorlbark — APPROVED** [review]
👍

File: .github/workflows/build.yaml (hunk 1 above)
Comment (rushtong) [reply .github/workflows/build.yaml]: Turns out I was very confused about the difference between `GITHUB_OUTPUT` and `GITHUB_ENV`. Using `GITHUB_OUTPUT` simplifies the PR quite a bit.

**Review by fboulnois — APPROVED** [review]

**Pull request merged**